action.yaml

```diff
@@ -14,14 +14,31 @@ inputs:
 runs:
   using: "composite"
   steps:
-    - name: install dependencies
-      run: mix deps.get
+    - run: sudo apt-get --assume-yes update 
+      shell: bash
+    - run: sudo apt-get --assume-yes install perl cpanminus
+      shell: bash
+    - run: perl -v
+      shell: bash
+    - run: cpanm --notest --sudo Dist::Zilla
+      shell: bash
+    - run: dzil --version
+      shell: bash
+    - name: install dependencies required for building
+      run: dzil authordeps --missing | cpanm --sudo --notest
       shell: bash
       working-directory: ${{ inputs.working-directory }}
-    - name: publish package
+    - name: build package
       run: |
-        mix hex.publish --yes
-      env:
-        HEX_API_KEY: ${{ inputs.hex-api-key }}
+        echo Building version $(cat ./VERSION)
+        dzil build
       shell: bash
       working-directory: ${{ inputs.working-directory }}
+    - name: upload to cpan
+      run: |
+        cpan-upload --user $CPAN_USER --password $CPAN_PASSWORD *.tar.gz
```

Comment:
I would exect there to be exactly 1 `.tar.gz` file, but this doesn't work if there are multiple for some reason. Based on the `dist.ini` file and the content of `VERSION` we know exactly which version we'll be pushing out...

my main point: this works for the Cucumber project, but isn't generic.

Reply:
Is there a way we could construct the filename from calls to `cpan`/`cpanm`/`dzil` or somesuch, to read the `dist.ini`?

Reply:
I checked, but can't find how that would be. It's in the file "META.yml", but I think that only exists inside the tar file.

Reply:
So there's no programatic way to get the version number that `dzil` can read from the dist.ini?

Reply:
well, https://metacpan.org/pod/Config::INI::Reader is the dependency dzil uses to read `dist.ini`. If the version number is in `dist.ini` then we ought to be able to use that to extractit. I meant that we want to be handed the full name of the `.tar.gz` file: basically ask dzil what the name of the file will be. That would work even when dzil tries to extract the version number from the main module.

Reply:
maybe we could read it from the output of `dzil build`?

Reply:
```
➜  perl git:(make-it-work) dzil build
[DZ] beginning to build Cucumber-Test-Release-Automation
[DZ] guessing dist's main_module is lib/Cucumber/Test/Release/Automation.pm
[DZ] writing Cucumber-Test-Release-Automation in Cucumber-Test-Release-Automation-v0.0.0
[DZ] building archive with Archive::Tar; install Archive::Tar::Wrapper 0.15 or newer for improved speed
[DZ] writing archive to Cucumber-Test-Release-Automation-v0.0.0.tar.gz
[DZ] built in Cucumber-Test-Release-Automation-v0.0.0
```

Reply:
Done in https://github.com/cucumber/action-publish-cpan/pull/2/commits/c9a51037022e5f8c56a1853e00df229ded32305a